src/components/SignUp.js

```diff
@@ -0,0 +1,172 @@
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import styled, { ThemeProvider } from "styled-components";
+import axios from "axios";
+
+import StyledButton from "../styles/commonStyle";
+import theme from "../styles/theme";
+
+const SignUp = () => {
+  const [message, setMessage] = useState("");
+  const { signUpInfo } = useSelector(state => state.signUp);
+  const navigate = useNavigate();
+
+  const handleSaveButtonClick = async event => {
+    event.preventDefault();
+
+    const nickname = event.target.nickname.value;
+    const contact = event.target.contact.value;
+    const birthday = event.target.birthday.value;
+    const occupation = event.target.occupation.value;
```

Comment:
이 부분은 위에서 각 input 들을 구조분해할당 해준 뒤 값을 할당해줘도 좋을 것 같기도 한데 어떻게 생각하세요~?

Reply:
성훈님 말씀대로 구조분해 할당도 좋을 것 같긴한데요
구조분해 할당을 한다면

```suggestion
    const { nickname, contact, birthday, occupation } = event.target;
```
이렇게 받아오고 각 value를 따로 가져와야해서 서버에 데이터를 넘겨줄 때

```
    const { data } = await axios.post("/auth/sign-up", {
      user: {
        email: signUpInfo.email,
        profileUrl: signUpInfo.profileUrl,
        nickname: nickname.value,
        contact: contact.value,
        birthday: birthday.value,
        occupation: occupation.value,
      },
    });
```
이렇게 넘겨줘야해서 구조분해할당을 하지 않았는데,, 구조분해 할당을 해도

```
    const { data } = await axios.post("/auth/sign-up", {
      user: {
        email: signUpInfo.email,
        profileUrl: signUpInfo.profileUrl,
        nickname,
        contact,
        birthday,
        occupation,
      },
    });
```
이렇게 깔끔하게 넘겨줄 수 있는 방법이 있을까요-?

Reply:
아아 네네! 서버에 넘겨줄때는 현재처럼 생략가능한 문법으로 보내주는게 좋은 것 같아요! 
저는 해당 코드라인을 이렇게 변경하는 건 어떨까 싶어서 여쭤봤습니다ㅎㅎ

```suggestion
    const { nickname, contact, birthday, occupation } = event.target;
    const nickname = nickname.value;
    const contact = contact.value;
    const birthday = birthday.value;
    const occupation = occupation.value;
```

Reply:
오오 네 말씀하신대로 수정해보겠습니다-

Reply:
앗 네 그렇네요 제가 생각을 잘못 한 것 같아요 영진님~
답변감사합니다~